src/thunderbots/software/ai/hl/stp/stp.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include <random>
+
+#include "ai/hl/hl.h"
+#include "ai/hl/stp/play/play.h"
+#include "ai/intent/intent.h"
+
+/**
+ * The STP module is an implementation of the high-level logic Abstract class, that
+ * uses the STP (Skills, Tactics, Plays) framework for its decision making.
+ */
+class STP : public HL
+{
+   public:
+    /**
+     * Creates a new High-Level logic module that uses the STP framework for
+     * decision-making.
+     *
+     * @param random_seed The random seed used for STP's internal random number generator.
+     * Defaults to the number of nanoseconds since the Linux epoch
+     */
+    explicit STP(
+        long random_seed = std::chrono::system_clock::now().time_since_epoch().count());
+
+    std::vector<std::unique_ptr<Intent>> getIntentAssignment(const World& world) override;
+
+    /**
+     * Given a list of tactics, assigns each available friendly Robot to the tactic it
+     * should run, and returns a pairing of Robots to Tactics
+     *
+     * The priority determines what Tactics will not
+     * be assigned if there are not enough robots on the field to assign them all. For
+     * example, if a Play returned 6 Tactics but there were only 4 robots on the field
+     * at the time, only the first 4 Tactics in the vector would be assigned to robots
+     * and run.
+     *
+     * @param world The state of the world, which contains the friendly Robots that will
+     * be mapped to a Tactic
+     * @param tactics The list of tactics that should be run (and paired with a Robot)
+     * @return A list of pairs, where each pair contains a Robot and the Tactic that the
+     * Robot will run
+     */
+    std::vector<std::shared_ptr<Tactic>> assignRobotsToTactics(
```

Comment:
This returns a vector of `Tactic` not a pair?

Reply:
Yes. It returns the list of tactics, with robots assigned to them. Would you prefer returning a vector of pairs, and then explicitly doing the assignment  outside this function?

Reply:
No, I'm find with this, but it doesn't match the `@return` in the javadoc

Reply:
Ah I see my `@return` comment was out of date. I'm going to change it to return a list of pairs anyway since I think that's more explicit.